deepspeed/accelerator/abstract_accelerator.py

```diff
@@ -0,0 +1,198 @@
+import abc
+from abc import ABC
+
+
+class DeepSpeedAccelerator(ABC):
+    def __init__(self):
+        self._name = None
+        self._communication_backend_name = None
+        self.BFloat16Tensor = None
+        self.ByteTensor = None
+        self.DoubleTensor = None
+        self.FloatTensor = None
+        self.HalfTensor = None
+        self.IntTensor = None
+        self.LongTensor = None
```

Comment:
Are these values required to be defined for all inheriting classes? It could be useful to use `@abstractproperty` for them, so we ensure all accelerators define them.

Reply:
`_name` and `_communication_backend_name` already has get function defined so they should not be defined as `@abstractproperty`.   The rest tensor type class should be defined for all inheriting class and can be defined as `@abstractproperty`.

Note: the following names are not actually used by DeepSpeed, they are defined for compleness:
`BFloat16Tensor`, `DoubleTensor`, `HalfTensor`

Reply:
Updated according to comments